### PR TITLE
Some interpolation related fixes

### PR DESCRIPTION
**Pull request by alufolie91**
Most are from RR
we had some of this in saturn before tho this time its minus moving R_UpdateViewInterpolation into G_Ticker as that caused issues in netgames
This def needs a bit of testing esp in netgames and under poor netconditions, as the stuff we had before sometimes caused jitter on mobjs
but this would fix some underlying issues with how view interpolation works
The code in p_telpt.c was supposed to use P_SetOrigin but that may not be netsafe so i set the interp values manually
also fixes https://github.com/Indev450/SRB2Kart-Saturn/issues/140

### CODE REVIEW

**Comment by alufolie91** [comment]
im a bit unsure actually about the stuff in y_inter mainly cause i dont really understand whats even going on there lol, would be cool if you would look at that @Indev450 X)

**Comment by Indev450** [comment]
I don't know what's going on in y_inter either x)
I dunno what that jitter does but i think changing it like that will change behaviour, since it will not be reset like it used to. Which is probably fine but dunno if that's intended

File: src/r_fps.c [hunk 1]
```diff
@@ -207,21 +209,27 @@ void R_UpdateViewInterpolation(void)
...
 	}
+
+	last_view_update = I_GetTime();
 }
 
 void R_ResetViewInterpolation(UINT8 p)
 {
+	// Wait an extra tic if the interpolation state hasn't
+	// updated yet.
+	int t = last_view_update == I_GetTime() ? 1 : 2;
```
Comment (Indev450) [inline src/r_fps.c]: This probably should have parentheses around == (`(last_view_update == I_GetTime()) ? 1 : 2;`). Not sure about order of operations here but just to be safe

File: src/r_fps.c [hunk 2]
```diff
@@ -207,21 +209,27 @@ void R_UpdateViewInterpolation(void)
...
+
 	if (p == 0)
 	{
 		UINT8 i;
 		for (i = 0; i < MAXSPLITSCREENPLAYERS; i++)
 		{
-			oldview_invalid[i]++;
+			oldview_invalid[i] = t;
```
Comment (Indev450) [inline src/r_fps.c:227]: Shouldn't that be `+=` instead of `=`?

File: src/r_fps.c [hunk 3]
```diff
@@ -207,21 +209,27 @@ void R_UpdateViewInterpolation(void)
...
 		}
 	}
 	else
 	{
-		oldview_invalid[p - 1]++;
+		oldview_invalid[p - 1] = t;
```
Comment (Indev450) [inline src/r_fps.c:232]: Same as above x)

File: src/r_fps.c (hunk 2 above)
Comment (alufolie91) [reply src/r_fps.c:227]: well lol https://github.com/KartKrewDev/RingRacers/blob/dd45f3cd61a6afa83cac876a0ae7d43fa2d1ab43/src/r_fps.c#L175

File: src/r_fps.c (hunk 1 above)
Comment (alufolie91) [reply src/r_fps.c]: added some there

File: src/r_fps.c (hunk 2 above)
Comment (alufolie91) [reply src/r_fps.c:227]: actually testing with += currently

Comment (alufolie91) [reply src/r_fps.c:227]: ye uh lmao its intentional so `"do not add extra time for subsequent calls"`

**Pull request merged**